### DISCUSSION TITLE
Multi-Agent Collaboration Feature

### DISCUSSION THREAD

**Discussion by 9cren** [langgenius/dify#4676] [Suggestion]
I am writing to request the addition of a multi-agent collaboration feature in future versions of the Dify project. Currently, the agent functionality is somewhat limited. It would be highly beneficial to incorporate capabilities similar to those found in the Coze platform, where multiple agents can work together seamlessly to accomplish tasks. This enhancement would significantly improve the utility and flexibility of Dify for various applications. Thank you for considering this request.

**Comment by patryk20120** [comment]
You can easily implement this using chatflow and some kind of supervisor agent.

**Reply by computersrmyfriends** [reply]
> You can easily implement this using chatflow and some kind of supervisor agent.

I understand the chatflow part but What kind of supervisor agent do you mean? I can't find a supervisor agent option in dify.

**Reply by patryk20120** [reply]
Well a supervisor is just a pattern, where you use LLM to decide what tool it needs to use

**Reply by computersrmyfriends** [reply]
> Well a supervisor is just a pattern, where you use LLM to decide what tool it needs to use

I already have created a workflow within dify that uses multiple LLM nodes to create a long article with multiple sections. I am combining the outputs of multiple LLMs using the Jinja template node to create the one big article.
I tried creating an agent but how do I make it supervise and rewrite the multi page document that is generated by the workflow? Currently, the article appears as responses from multiple LLMs.

**Reply by hponiang** [reply]
I have a very similar requirement. A report consists of four parts, which need to be completed by multiple agents, and then the contents need to be merged at the end.

**Reply by computersrmyfriends** [reply]
I am able to concatenate the text using the template node however, is it possible for Dify to rewrite the whole text into one single large article so that it doesn't appear like the concatenation of multiple AI responses?

**Reply by hponiang** [reply]
> 我能够使用模板节点连接文本，但是，Dify 是否可以将整个文本重写为一篇大文章，这样它就不会看起来像是多个 AI 响应的串联？

"I am currently thinking of using LangChain to call multiple agent interfaces. Do you have a better approach?"
